var/spack/repos/builtin/packages/automake/package.py

```diff
@@ -31,20 +31,13 @@ class Automake(AutotoolsPackage, GNUMirrorPackage):
     executables = ['automake']
 
     @classmethod
-    def determine_spec_details(cls, prefix, exes_in_prefix):
-        exe_to_path = dict(
-            (os.path.basename(p), p) for p in exes_in_prefix
-        )
-        if 'automake' not in exe_to_path:
+    def determine_version(cls, exe):
+        if os.path.basename(exe) != 'automake':
             return None
```

Comment:
I would like to remove the necessity of this line in every function. Either make `executables` be `'^automake$'` or add that by default and let users add `'.*automake.*'` if they need to.

Reply:
I'll go with `'^automake$'`